test/unit/signal.spec.js

```diff
@@ -1,36 +1,167 @@
-import Signal from '../../lib/api/signal'
+import { MemoizedSignal, Signal } from '../../lib/api/signal'
 import {noop} from '../helpers'
 
-describe(`Signal`, () => {
-  describe(`constructor`, () => {
-    it(`creates a Signal instance`, () => {
+describe('MemoizedSignal', function () {
```

Comment:
Duplicates all the tests from `Signal`. We should either remove those or parametrize them over the constructor and run them for both classes.